src/app/cake/cake.service.ts

```diff
@@ -0,0 +1,71 @@
+import Cake from './models/cake.model';
+// Memory data for mimic endpoints functionality...
+let cakes: Cake[] = [];
+
+/**
+ * This service will return the list of cakes
+ */
+export const getCakes = () => {
+  return new Promise<Cake[]>((resolve, reject) => {
+    setTimeout(() => {
+      // business logic
+      resolve(cakes);
+    }, cakes.length * 100);
+  });
+};
+
+/**
+ * This service will find a Cake  by name or ID
+ * @param cakeName cake name/id to find and return it
+ */
+export const getCake = (cakeName: string) => {
+  return new Promise<Cake | null>((resolve, reject) => {
+    setTimeout(() => {
+      // business logic
```

Comment:
Business logic should be added in the class :D

Reply:
Thanks, I got it, I'll do it that way. Regarding to that TO-DO comments, those are my own markers for my WIP haha.